airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -539,11 +540,24 @@ def load_string(
         :param acl_policy: The string to specify the canned ACL policy for the
             object to be uploaded
         :type acl_policy: str
+        :param compression: Type of compression to use, currently only gzip is supported.
+        :type compression: str
         """
         encoding = encoding or 'utf-8'
 
         bytes_data = string_data.encode(encoding)
+
+        # Compress string
+        if compression not in (None, 'gzip'):
+            raise NotImplementedError(
+                "Received {} compression type. String "
+                "can currently be compressed in gzip only.".format(compression)
+            )
```

Comment:
Maybe we could automatically create the error message based on available compressions if we store these in a variable and read from it.

Reply:
Thanks for the suggestion, I've updated the PR. And also added one more test to check the exception raised. Can you please take a look again?